### PR TITLE
Replaced Joi with class-validator library for server-side validation

### PR DESCRIPTION
**Pull request by ericpera-msft**
Helium now utilizes class decorators to enforce server-side validation on POST requests for actors and movies.

### CODE REVIEW

**Comment by frodopwns** [comment]
Part of #145

File: src/utilities/validationUtilities.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { registerDecorator, ValidationArguments, ValidationOptions } from "class-validator";
+
+// Custom decorator that compares two properties in an object instance
```
Comment (frodopwns) [inline src/utilities/validationUtilities.ts]: not the proper doc string for auto generated docs...should contain param info

File: src/app/controllers/actor.ts [hunk 2]
```diff
@@ -95,29 +96,21 @@ export class ActorController implements interfaces.Controller {
...
-                    message: validation.error.details.map((err) => err.context.key === "textSearch"
-                        ? `"textSearch" must be equal to "${req.body.name.toLowerCase()}"`
-                        : err.message),
-                    status: 400,
-                });
-        }
+        const actor: Actor = Object.assign(Object.create(Actor.prototype),
+            JSON.parse(JSON.stringify(req.body)));
+
+        actor.isValid().then(async (errors) => {
```
[the remark this answers is not in the document]
Comment (ericpera-msft) [reply src/app/controllers/actor.ts]: I was torn between calling it validate or isValid. In the beginning, it made sense to call it isValid but then I changed my approach to what it is now and I agree that validate makes more sense with how this code turned out. Good call, I'll make the change.

File: src/app/controllers/actor.ts [hunk 3]
```diff
@@ -95,29 +82,21 @@ export class ActorController implements interfaces.Controller {
...
-            return res.send(400,
-                {
-                    message: validation.error.details.map((err) => err.context.key === "textSearch"
-                        ? `"textSearch" must be equal to "${req.body.name.toLowerCase()}"`
-                        : err.message),
-                    status: 400,
-                });
-        }
+        const actor: Actor = Object.assign(Object.create(Actor.prototype),
+            JSON.parse(JSON.stringify(req.body)));
```
Comment (frodopwns) [inline src/app/controllers/actor.ts]:
```
const actor: Actor = req.body;
```
?

File: src/app/controllers/movie.ts [hunk 4]
```diff
@@ -86,30 +74,22 @@ export class MovieController implements interfaces.Controller {
...
-                {
-                    title: req.body.title !== undefined ? req.body.title.toLowerCase() : "",
-                },
-            });
-
-        // result.error === null -> valid
-        if (validation.error === null) {
-            const result = await this.cosmosDb.upsertDocument(database, collection, req.body);
+        const movie: Movie = Object.assign(Object.create(Movie.prototype),
+            JSON.parse(JSON.stringify(req.body)));
```
Comment (frodopwns) [inline src/app/controllers/movie.ts]:
```
const movie: Movie = req.body;
```

Any problem with this?

**Pull request merged**